src/bio/bio_wal.c

```diff
@@ -663,7 +663,7 @@ fill_trans_blks(struct bio_meta_context *mc, struct bio_sglist *bsgl, struct ume
 static inline uint64_t
 off2lba(struct wal_super_info *si, unsigned int blk_off)
 {
-	return (blk_off + WAL_HDR_BLKS) * si->si_header.wh_blk_bytes;
+	return (uint64_t)(blk_off + WAL_HDR_BLKS) * si->si_header.wh_blk_bytes;
```

Comment:
Is that the only place where integer overflow can be? If it happens here, could "blks * blk_bytes" also be overflown at line  958?
        bio_iov_set(&bsgl->bs_iovs[0], addr, blks * blk_bytes);

Reply:
The size of WAL transaction is quite limited (we currently set the max wal tx size as 8MB, it could be bumped if necessary, but I don't think it will exceed 4GB), so this one won't overflow.